### ISSUE TITLE
add Material

### ISSUE THREAD

**Issue by nathanfranklin** [simphony/simphony-common#227]
In relation with #226 (https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/MaterialRelations_proposal), a `Material` class has been defined by the SSB as the following:

```
class Material(object):
    """A class that represents a material.

    Class describes a material and its data.

    Attributes
    ----------
    description : str
        textual description of the material
    uid : uuid.UUID
        the id of the material.
    data : DataContainer
        data associated with this material

    """
```
And the the `uid` of materials will be used like the following:
```
material_foo = Material(name="foo")
material.data[CUBA.MASS] = 1.0  # this is a common mass for entities of this material type

p = Particle()
p.data[CUBA.MATERIAL] = material_foo.uid

engine.SD.add_material(material_foo)

dataset = engine.get_dataset("some_name")
dataset.add_particles([p])
```

This requires: 
- [ ] Adding a new attribute (CUBA.MATERIAL) with type that can represent the UUID
- [ ] Adding `Material` class (plus tests and documentation)
- [ ] Add `SD` object to contain these materials (i.e. `add_material`, `get_material`, `delete_material`, `update_material`) (plus tests and documentation)

**Comment by ahashibon** [comment]
It would be better at this stage to hold on with the SD component, as it may not be needed at this stage. Implementing the Material in the engine now is sufficient. This is due to ongoing discussions in the Standards Body regarding the form of the CUDS.

**Comment by nathanfranklin** [comment]
@ahashibon , sounds good.  I'll update #237 so that the methods to work with materials are added to ABCModeling engine i.e:
```
material_foo = Material(name="foo")
material.data[CUBA.MASS] = 1.0 
engine.SD.add_material(material_foo)
```

**Comment by nathanfranklin** [comment]
>It would be better at this stage to hold on with the SD component, as it may not be needed at this stage. Implementing the Material in the engine now is sufficient. This is due to ongoing discussions in the Standards Body regarding the form of the CUDS.  

@ahashibon   Should we also hold off on the SP component in  #226 as well?   Should we stop extending the SP object to contain these material relations and instead just add the material-relations methods (i.e. add_material_relation, get_material_relation, delete_material_relation, update_material_relation) to `ABCModelingEngine`

**Comment by ahashibon** [comment]
yes, hold on with the SP. Follow update here: https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/New_Design:_Towards_1.6#Example:_Mixture_model_.28without_.E2.80.98fluid.E2.80.99_concept.2C_DEM_coupling.29

**Comment by nathanfranklin** [comment]
Closed by #237
